elasticache/common.go

```diff
@@ -0,0 +1,44 @@
+package elasticache
+
+import (
+	"github.com/aws/aws-cdk-go/awscdk/v2/awsec2"
+	"github.com/aws/aws-cdk-go/awscdk/v2/awselasticache"
+	"github.com/aws/constructs-go/constructs/v10"
+	"github.com/aws/jsii-runtime-go"
+)
+
+// GetPrivateSubnetGroup returns a new subnet group for the given elasticache stack
+func GetPrivateSubnetGroup(scope constructs.Construct, name *string, vpc awsec2.IVpc) awselasticache.CfnSubnetGroup {
+	privateSubnetIDs := make([]*string, 0)
+
+	for _, subnet := range *vpc.PrivateSubnets() {
+		privateSubnetIDs = append(privateSubnetIDs, subnet.SubnetId())
+	}
+
+	subnetGroup := awselasticache.NewCfnSubnetGroup(scope, name, &awselasticache.CfnSubnetGroupProps{
+		Description:          jsii.String("Acorn created Elasticache subnets"),
+		CacheSubnetGroupName: name,
+		SubnetIds:            &privateSubnetIDs,
+	})
+
+	return subnetGroup
+}
+
+// GetAllowAllVPCSecurityGroup returns a security group that allows traffic to and from the Elasticache cluster
+func GetAllowAllVPCSecurityGroup(scope constructs.Construct, name *string, vpc awsec2.IVpc, port int) awsec2.SecurityGroup {
```

Comment:
Is this similar to the rds function? If so, lets move it to libs/common.

Reply:
Moved into `libs/common` and replaced calls in RDS and Redis to use the common implementation in the latest commit.